alns/accept/NonLinearGreatDeluge.py

```diff
@@ -0,0 +1,106 @@
+    a non-linear updating scheme, see the ``_update`` method for more details.
+    Moreover, candidate solutions that improve the current solution are always
+    accepted.
+
+    The implementation is based on the description in [2].
+
+    Parameters
+    ----------
+    alpha
+        Factor used to compute the initial threshold
+    beta
+        Factor used to update the threshold
+    gamma
+        Facto1 used to update the threshold
+    delta
+        Factor used to update the threshold
+
+    References
+    ----------
+    [1]: Dueck, G. New optimization heuristics: The great deluge algorithm and
+         the record-to-record travel. *Journal of Computational Physics* (1993)
+         104 (1): 86-92.
+    [2]: Landa-Silva, D., & Obit, J. H. Great deluge with non-linear decay rate
+         for solving course timetabling problems. *4th international IEEE
+         conference intelligent systems* (2008) Vol. 1: 8-11.
+    [3]: Santini, A., Ropke, S. & Hvattum, L.M. A comparison of acceptance
+         criteria for the adaptive large neighbourhood search metaheuristic.
+         *Journal of Heuristics* (2018) 24 (5): 783–815.
+    """
+
+    def __init__(self, alpha: float, beta: float, gamma: float, delta: float):
+        super().__init__(alpha, beta)
+
+        if gamma <= 0 or delta <= 0:
+            raise ValueError("Gamma and delta must be non-negative.")
+
+        self._gamma = gamma
+        self._delta = delta
+
+    @property
+    def gamma(self):
+        return self._gamma
+
+    @property
+    def delta(self):
+        return self._delta
+
+    def __call__(self, rnd, best, curr, cand):
+        if best.objective() == 0:
+            raise ValueError("Initial solution cannot have zero value.")
+
+        if self._threshold is None:
+            self._threshold = self._alpha * best.objective()
+
+        res = cand.objective() < self._threshold
+
+        if not res:
+            res |= cand.objective() < curr.objective()  # Accept improving
+
+        self._threshold = self._update(best, curr, cand)
+
+        return res
+
+    def _update(self, best, curr, cand):
+        """
+        Return the updated threshold value.
+
+        First, the relative gap between the candidate solution and threshold
+        is computed. If this relative gap is less than ``beta``, then the
+        threshold is linearly increased (involving the ``gamma`` parameter).
+        Otherwise, the threshold is exponentially decreased (involing the
+        ``delta`` parameter).
+        """
+
+        rel_gap = (self._threshold - cand.objective()) / self._threshold
+
+        if rel_gap < self._beta:
+            res = (
+                self.gamma * abs(cand.objective() - self._threshold)
+                + self._threshold
+            )
+        else:
+            res = (
+                self._threshold * math.exp(-self.delta * best.objective())
+                + best.objective()
+            )
```

Comment:
Refactor: can you compute this in multiple lines, so we get rid of the brackets?

E.g.
```python
if ...:
  term1 = self.gamma * abs(...)
  term2 = self._threshold
else:
  term1 = self._threshold * math.exp(...)
  term2 = best.objective()

return term1 + term2
```